### PR TITLE
feat: add native resolver for did:key

### LINKED ISSUE #352: Add did:key native resolver
Add did:key native resolver

### PR DESCRIPTION
**Pull request by mirceanis**
This uses [did-key.js](https://github.com/transmute-industries/did-key.js) from transmute-industries

fixes #352

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1529,6 +1529,30 @@
...
+  integrity sha512-r6ed3N9xO+7AlO6FF2rVksH+HjfLwxZ9eKWHg3saTi5hzhJyehE3Qahj+goddgYD6emzwzrVbawNZDTttMQUKg==
+  optionalDependencies:
+    "@mattrglobal/node-bbs-signatures" "0.10.0"
+
+"@mattrglobal/bls12381-key-pair@^0.4.1-unstable.256739c":
+  version "0.4.1-unstable.f581696"
+  resolved "https://registry.yarnpkg.com/@mattrglobal/bls12381-key-pair/-/bls12381-key-pair-0.4.1-unstable.f581696.tgz#989c9b918baa1b26a2db6ac3a90115eb9d7f8c9f"
+  integrity sha512-EtOK60//gsmZjtzumdN08pfa/41tSpY1XNrwKhxq3fv0z3a7Tna2+HflnohhUQOGH7xpvpd+aUkzDm/OFpMsaA==
+  dependencies:
+    "@mattrglobal/bbs-signatures" "0.4.0"
```
Comment (rado0x54) [inline yarn.lock:1544]: Not completely sure why a signature package depends on a Key package. We should probably report that to Mattr (should be the other way round)

File: yarn.lock [hunk 2]
```diff
@@ -2075,6 +2152,110 @@
   resolved "https://registry.yarnpkg.com/@tootallnate/once/-/once-1.1.2.tgz#ccb91445360179a04e7fe6aff78c00ffc1eeaf82"
   integrity sha512-RbzJvlNzmRq5c3O09UipeuXno4tA1FE6ikOjxZK0tuxVv3412l64l5t1W5pj4+rJq9vpkm/kwiR07aZXnsKPxw==
 
+"@transmute/did-key-bls12381@^0.2.1-unstable.35":
+  version "0.2.1-unstable.35"
+  resolved "https://registry.yarnpkg.com/@transmute/did-key-bls12381/-/did-key-bls12381-0.2.1-unstable.35.tgz#5b0f64f50aca8a0cb615db11e29c9777c837f7af"
+  integrity sha512-P6+EWeO3QShR7mEK8tPfUotYrBD1QKi7G6lGEVhFQI5hVoqK7ElQxhiWBYh789Xm60AWRwcA+QhsYaJG1LmLeA==
+  dependencies:
+    "@mattrglobal/bbs-signatures" "^0.4.0"
```
Comment (rado0x54) [inline yarn.lock:2160]: Same here. Would not have expected that to be a dependency.

**Review by rado0x54 — APPROVED** [review]
Generally I feel did:key from transmute pulls in a lot of dependencies and since for a lot of our implementations we do not implement tree-shaking in any way it will get quite big in our deployments.

Not urgent now, but something that we might want to discuss in the future.

**Comment by mirceanis** [comment]
> Generally I feel did:key from transmute pulls in a lot of dependencies and since for a lot of our implementations we do not implement tree-shaking in any way it will get quite big in our deployments.

Yeah, that's true.
However, these are not core dependencies; so it's not an issue until you actually use the did:key resolver/provider package.

**Comment by mirceanis** [comment]
recreated in #458

**Pull request closed without merging**